src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/field_control_form.tsx

```diff
@@ -0,0 +1,276 @@
+  initialState?: ESQLControlState;
+  onCancelControl?: () => void;
+}
+
+export function FieldControlForm({
+  variableType,
+  initialState,
+  queryString,
+  esqlVariables,
+  cursorPosition,
+  onCreateControl,
+  onEditControl,
+  onCancelControl,
+  search,
+  closeFlyout,
+}: FieldControlFormProps) {
+  const suggestedVariableName = useMemo(() => {
+    const existingVariables = esqlVariables.filter((variable) => variable.type === variableType);
+
+    return initialState
+      ? `${initialState.variableName}`
+      : getRecurrentVariableName(
+          'field',
+          existingVariables.map((variable) => variable.key)
+        );
+  }, [esqlVariables, initialState, variableType]);
+
+  const [availableFieldsOptions, setAvailableFieldsOptions] = useState<EuiComboBoxOptionOption[]>(
+    []
+  );
+
+  const [selectedFields, setSelectedFields] = useState<EuiComboBoxOptionOption[]>(
+    initialState
+      ? initialState.availableOptions.map((option) => {
+          return {
+            label: option,
+            key: option,
+            'data-test-subj': option,
+          };
+        })
+      : []
+  );
+  const [formIsInvalid, setFormIsInvalid] = useState(false);
+  const [variableName, setVariableName] = useState(suggestedVariableName);
+  const [label, setLabel] = useState(initialState?.title ?? '');
+  const [minimumWidth, setMinimumWidth] = useState(initialState?.width ?? 'medium');
+  const [grow, setGrow] = useState(initialState?.grow ?? false);
+
+  const isControlInEditMode = useMemo(() => !!initialState, [initialState]);
+
+  useEffect(() => {
+    if (!availableFieldsOptions.length) {
+      const queryForFields = getQueryForFields(queryString, cursorPosition);
+      getESQLQueryColumnsRaw({
+        esqlQuery: queryForFields,
+        search,
+      }).then((columns) => {
+        setAvailableFieldsOptions(
+          columns.map((col) => {
+            return {
+              label: col.name,
+              key: col.name,
+              'data-test-subj': col.name,
+            };
+          })
+        );
+      });
+    }
+  }, [availableFieldsOptions.length, variableType, cursorPosition, queryString, search]);
+
+  useEffect(() => {
+    const variableExists =
+      esqlVariables.some((variable) => variable.key === variableName.replace('?', '')) &&
+      !isControlInEditMode;
+
+    setFormIsInvalid(!selectedFields.length || !variableName || variableExists);
+  }, [esqlVariables, isControlInEditMode, selectedFields.length, variableName]);
+
+  const onFieldsChange = useCallback((selectedOptions: EuiComboBoxOptionOption[]) => {
+    setSelectedFields(selectedOptions);
+  }, []);
+
+  const onVariableNameChange = useCallback(
+    (e: { target: { value: React.SetStateAction<string> } }) => {
+      let text = String(e.target.value).replace('?', '');
+      if (text.charAt(0) === '_') {
+        text = text.substring(1);
+      }
+      setVariableName(text);
+    },
+    []
+  );
```

Comment:
There is no validation on the variable name and it cause both ESQL exceptions and also unfixable lens configuration until you don't remove completely the control.
If you try to remove the wrong variable from the ESQL query you are not allow to save it and the error is unrecoverable if you don't remove the control from the dashboard first.

This happens for example if you use a variable name that contains white spaces or that starts with a `/`.
You should harden the validation and blocks or disallow using special characters, or, if possible, escape it.

Reply:
Thanx Marco, done here https://github.com/elastic/kibana/pull/202875/commits/031386ff58eb8c26ce0564b2c9c16451cff9fa2c and added a test too